packages/ember-data/lib/system/model/model.js

```diff
@@ -781,7 +782,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
-  }
+  },
+
+  setId: Ember.observer('id', function () {
```

Comment:
ug the internal flow here seems very strange.

It really feels like it should be:

```js

model ... {
  get id()   { return this.internalModel.id; }
  set id(id) { this.internalModel.id = id; }
}
```
```
internalModel .. {
  set id(id) {
    if (this._id === id) { return; }

    this.store.moveToId(this, id);
    if (this.model) {
      this.model.notifyPropertyChange('id');
    }
  },

   get id() {
     return this._id;
   }
}
```